src/core/qgscoordinatereferencesystem.cpp

```diff
@@ -1398,8 +1398,8 @@ bool QgsCoordinateReferenceSystem::readXml( const QDomNode &node )
         //make sure the map units have been set
         setMapUnits();
 
-        //@TODO this srs needs to be validated!!!
-        d->mIsValid = true; //shamelessly hard coded for now
+
+        d->mIsValid = isValid();
```

Comment:
I don't think this line does anything at all.. given that isValid() returns d->mIsValid ;)